fix: UID/GID build time, handle existing GID

pollomarzo opened the pull request:
closes #85

pollomarzo committed:
fix: UID/GID build time, handle existing GID

closes #85

francesco-frosolini commented:
'=> ERROR [base  7/13] RUN mkdir /sim && chown simuser:simuser /sim                    0.3s
------
 > [base  7/13] RUN mkdir /sim && chown simuser:simuser /sim:
0.264 chown: invalid group: 'simuser:simuser'
------
Dockerfile:72'

I built using --no-cache

pollomarzo committed:
fix: remove GID from chown

it'll apply primary group anyway.
i took the chance to refactor usages of "simuser" out of the dockerfile
not 100% sure desired effect and no consequences... we'll see

pollomarzo commented:
lmk when you've tested and can confirm it works!

pollomarzo commented:
so from this output, the build works successfully, so it could be [this](https://github.com/near-nes/controller/blob/fa2bef1179fe48d97cb5a20ea827d441fb07fd46/docker/entrypoint.sh#L39) line that breaks. maybe `$TARGET_DIR` is not set or similar? could you check this?
we can also remove the whole group checking section, we're not using groups anyway, and they make the permission matching uselessly complicated. we should just check UID and proceed with that, but that's secondary

pollomarzo committed:
fix: remove GID matching

we don't use groups anyway, if we want it in the future we can include it again, but removing it simplifies things (groupname vs group id, does the group exist, what is it used for...)

francesco-frosolini commented:
correct: `$TARGET_DIR` is not set

```

root@fe49cbb9d2c4:/sim/controller# env | grep INSTALL_DIR
NEST_INSTALL_DIR=/sim/install/nest
INSTALL_DIR=/sim/install
NRP_INSTALL_DIR=/sim/install/nrp
OMPI_INSTALL_DIR=/sim/install/openmpi
NRP_DEPS_INSTALL_DIR=/sim/install/nrp_deps
MUSIC_INSTALL_DIR=/sim/install/music
root@fe49cbb9d2c4:/sim/controller# env | grep TARGET_DIR 
```

pollomarzo commented:
@francesco-frosolini did you have a chance to test this?

francesco-frosolini commented:
CONTROLLER_DIR is set by an ENV statement in the dockerfile. The only explanation I came up with for why TARGET_DIR is not set, is that [this](https://github.com/near-nes/controller/blob/ccbbe0c3a5508adb1842413393f104bdbddcc4e6/docker/entrypoint.sh#L11) line does not actually do anything, and it only matters for TARGET_DIR because the other env variables retain the differente name. Then, because it is not set, [this](https://github.com/near-nes/controller/blob/fa2bef1179fe48d97cb5a20ea827d441fb07fd46/docker/entrypoint.sh#L39) line breaks the UID/GID check.

Last time I tested, running export didn't do anything. I can also try to change the variable names but there must be something we are missing.

pollomarzo committed:
fix(docker): require UID/GID env vars, never stat

in bind mounts on docker for mac, stat returns 0:0. took this chance to remove it, and mandate that UID/GID are passed as input (.env is suitable). chown is skipped in HPC mode (maybe we'll split HPC/PROD down the line; no point using all that time just to allow `pip install`).

francesco-frosolini commented:
Looks good to me (I am testing on a 2019 i5 and 8Gb of RAM as well). Maybe it's because I built on top of a cached image from yesterday  (unsure if it was 33edf6f), I will try again with the --no-cache flag and update you

```shell
Switching to user simuser (UID: 501) and executing command: bash
----------------------------------------
Running passed command: bash
simuser@10801bd87316:/sim/controller$
```
